__tests__/app.test.js

```diff
@@ -85,26 +85,120 @@ describe("GET /api/reviews/:review_id", () => {
+  test("200: should return an array of objects, each with properties (comment_id, votes, created_at, author, body, review_id)", () => {
+    return request(app)
+      .get("/api/reviews/2/comments")
+      .expect(200)
+      .then(({ body: { comments } }) => {
+        expect(comments.length).toBe(3);
+        comments.forEach((comment) => {
+          expect(comment).toEqual(
+            expect.objectContaining({
+              comment_id: expect.any(Number),
+              votes: expect.any(Number),
+              created_at: expect.any(String),
+              author: expect.any(String),
+              body: expect.any(String),
+              review_id: expect.toBeInteger(2),
+            })
+          );
+        });
+      });
+  });
+  test("200: returned array should be sorted by created_at, descending", () => {
+    return request(app)
+      .get("/api/reviews/2/comments")
+      .expect(200)
+      .then(({ body: { comments } }) => {
+        expect(comments).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+  test("404: not found if no comments with the specified review_id", () => {
+    return request(app)
+      .get("/api/reviews/200/comments")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("not found");
+      });
+  });
+  test("400: bad request if review_id not valid data type", () => {
+    return request(app)
+      .get("/api/reviews/pineapple/comments")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("bad request");
+      });
+  });
+});
+
+describe('POST /api/reviews/:review_id/comments', () => {
+    test("201: responds with posted comment if succesful", () => {
+        const newComment = { username: 'dav3rid', body: 'life changing'}
+        return request(app)
+        .post("/api/reviews/1/comments")
+        .send(newComment)
+        .expect(201)
+        .then(({body: {comment}}) => {
+            expect(comment).toEqual({
+                comment_id: 7,
+                author: 'dav3rid',
+                body: 'life changing',
+                created_at: expect.any(String),
+                review_id: 1,
+                votes: 0
+            })
+        })
+    })
+    test("400: bad request if username does not exist in users table", () => {
+        const newComment = { username: 'fakeName123', body: 'life changing'}
+        return request(app)
+        .post("/api/reviews/1/comments")
+        .send(newComment)
+        .expect(400).then(({body : {msg}}) => {
+            expect(msg).toBe("bad request")
+        })
+    })
+    test("400: bad request if review_id does not exist in reviews table", () => {
+        const newComment = { username: 'dav3rid', body: 'life changing'}
+        return request(app)
+        .post("/api/reviews/1001/comments")
+        .send(newComment)
+        .expect(400).then(({body : {msg}}) => {
+            expect(msg).toBe("bad request")
+        })
```

Comment:
Could we please add a 400 test for missing required fields (no username or body property) ?